mtcpclient/calculate.go

```diff
@@ -38,26 +34,21 @@ func (gc GroupOfConnections) calculateMetricsReport(status tcpclient.ConnectionS
 				mr.total += item.GetTCPProcessingDuration()
 			}
 		}
+		mr.avg = mr.total / time.Duration(mr.numberOfConnections)
+		mr.stdDev = gc.calculateStdDev(mr)
 	}
-	mr.avg = mr.total / time.Duration(mr.numberOfConnections)
-	mr.stdDev = gc.calculateStdDev(status, mr)
 	return mr
 }
 
-func (gc GroupOfConnections) calculateStdDev(status tcpclient.ConnectionStatus, mr metricsCollectionStats) time.Duration {
+func (gc GroupOfConnections) calculateStdDev(mr metricsCollectionStats) time.Duration {
 	// TODO: passing the whole mr struct looks overkilling, given we only want a single value, the average, and maybe
 	//  we can actually use a version of the algorithm that calculates it (and the number of items)
-	var nItems int
 	var sd float64
-	for _, item := range gc {
-		if item.GetConnectionStatus() == status {
-			nItems++
-			sd += math.Pow(float64(item.GetTCPProcessingDuration())-float64(mr.avg), 2)
-		}
+	for _, item := range gc.connections {
+		sd += math.Pow(float64(item.GetTCPProcessingDuration())-float64(mr.avg), 2)
 	}
-	if nItems == 0 {
+	if mr.numberOfConnections == 0 {
```

Comment:
i recommend to change the order. This could be the first check in the function

Reply:
👍